File: run_net.lua
require 'nn'
require 'nngraph'
require 'mattorch'


function parse_args()
	cmd = torch.CmdLine()
	cmd:text('Options')
	-- general options:
	cmd:option('-model', '', 'Model Path')
	cmd:option('-inputs', '', 'Images Path')
    cmd:option('-res', '', 'Res Path')
	cmd:text()
	params = cmd:parse(arg)
	return params
end

params = parse_args()


model = torch.load(params.model)
input = mattorch.load(params.inputs)
in_im = input.img
in_im = in_im:view(1,in_im:size(1),in_im:size(2),in_im:size(3))
in_im = in_im:permute(1,2,4,3)
in_im:div(255.0):mul(2):add(-1) --Input Normalization
output = model:forward(in_im:double())
output:add(1):div(2):mul(255) --Output Denormalization
output = output:permute(1,2,4,3)
mattorch.save(params.res, output:double())

File: run_net.m
function [ res ] = run_net( img, net_path, save_path )
%run_net - Run the network in net_path
%   img - Input Image
%   net_path - The network path
%   save_path - Path for saving the network's output

matname = [save_path '.mat'];
img = double(img);
save(matname, 'img');
resname = [save_path  '-res.mat'];
system(['OMP_NUM_THREADS=1 th run_net.lua -model ' net_path ' -inputs ' matname ...
    ' -res ' resname ]);
net_res = load(resname);
delete(matname);
delete(resname);
res = net_res.x;


end

